lib/index.js

```diff
@@ -633,12 +670,25 @@ function preSave(next) {
  * internal
  * @param {Object} doc
  */
-async function postSave(doc) {
+async function postSave(doc, next) {
   if (doc && doc.esOptions) {
     const data = doc._mexp || {};
     const esOptions = doc.esOptions();
     delete doc._mexp;
+
     if (!esOptions.filter || esOptions.filter(doc)) {
+      if (esOptions.syncDependantIndexes && !data.wasNew) {
+        await syncDependantIndexes(
```

Comment:
We're try/catching this function in `postRemove`. Should we be doing the same here?